exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java

```diff
@@ -426,6 +445,73 @@ void getAuditorEmulatedNode() {
     }
   }
 
+  @Test
+  void timeServiceWorksInTestKit() {
+    FakeTimeProvider timeProvider = FakeTimeProvider.create(TIME);
+    try (TestKit testKit = TestKit.builder(EmulatedNodeType.VALIDATOR)
+        .withService(TestServiceModule.class)
+        .withTimeService(timeProvider)
+        .build()) {
+      // Commit two blocks for time oracle to prepare consolidated time
+      testKit.createBlock();
+      testKit.createBlock();
+      testKit.withSnapshot((view) -> {
+        TimeSchema timeSchema = TimeSchema.newInstance(view);
+        Optional<ZonedDateTime> consolidatedTime = timeSchema.getTime().toOptional();
+        assertThat(consolidatedTime).contains(TIME);
+
+        // Check that validatorsTimes contains one exactly entry with TestKit emulated node's
+        // public key and time provider's time
+        checkValidatorsTimes(timeSchema, testKit);
+        return null;
```

Comment:
I see withSnapshot is often used with Consumers, not Functions (unlike
`Node#withSnapshot`). We can't have them as overloaded parameters if I remember correctly
(there might be ambiguities for the compiler). Can anyone think of a better name?
- `withSnapshot(Function)` vs `verifyWithSnapshot(Consumer)` (we are testkit, aren't we?)
- `apply(Function)` vs `consume(Consumer)` (but won't apply give a false impression we modify
the state?)
- `testkit.snapshot().apply(Function)` vs `testkit.snapshot().consume(Function)` (might be too wordy, but more easily discoverable in testkit api than plain apply/consume).

I'd submit an issue to explore the options.

Reply:
I'd go with `withSnapshot(Function)` and `verifyWithSnapshot(Consumer)`. I don't think a separate issue is necessary, as there is no need in additional research.

Reply:
As it is a separate change in the public API of Testkit and the best option is TBD, I'd add an issue.